aws/resource_aws_security_group_rule.go

```diff
@@ -23,6 +24,25 @@ func resourceAwsSecurityGroupRule() *schema.Resource {
 		Read:   resourceAwsSecurityGroupRuleRead,
 		Update: resourceAwsSecurityGroupRuleUpdate,
 		Delete: resourceAwsSecurityGroupRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				importParts, err := validateSecurityGroupRuleImportString(d.Id())
+				if err != nil {
+					return nil, err
+				}
+				if err := populateSecurityGroupRuleFromImport(d, importParts); err != nil {
+					return nil, err
+				}
+				err = resourceAwsSecurityGroupRuleRead(d, meta)
```

Comment:
The `Read` function is automatically called, so this can be removed and simply: `return []*schema.ResourceData{d}, nil` 😄

Reply:
👍